### PR TITLE
resistor-color: add new exercise

### LINKED ISSUE #1458: Proposal for new exercise: ResistorColor (Resistor Series #1)
Resistors have color bands. Each color represents a number. Resistor can have up to 6 bands, and together they describe the resistance value in Ohms, together with the tolerance and the temperature coefficient.  

image

We're planning to have a series of Resistor-* exercises, that are building up to decoding 4-6 band resistor color codes.

1) The new exercise ResistorColor
 The first and most basic one is just reading the (first) band's color from an array and return its numeric value. 
schermafbeelding 2019-02-10 om 13 15 39

See PR #1457 

2) This is supposed to be part of a series of Resistor related exercises, in ascending complexity:
- Reading the first two bands and combine them to a number 
- Reading TriBand, and returning a value in Ω (Ohms) or K Ω 
- Reading 4 to 6 bands, returning the value in Ohms, the tolerance and the temperature coefficient. In those resistors, the meaning of the 3rd and 4th band is depending on the total number of bands. 
Coming soon!

### PR DESCRIPTION
**Pull request by ErikSchierboom**
The main purpose of this exercise is to add a very simple, array-based exercise that can serve as a very gentle introduction to arrays. As such, it has no error handling and the array is fixed.

There will be an issue opened relating to this PR later.

Closes #1458

### CODE REVIEW

File: exercises/resistor-color/description.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+- Yellow: 4
+- Green: 5
+- Blue: 6
+- Violet: 7
+- Grey: 8
+- White: 9
+
+Mnemonics map the colors to the numbers, that, when stored as an array, happen to map to their index in the array: Better Be Right Or Your Great Big Values Go Wrong.
+
+More information on the color encoding of resistors can be found [here.](https://en.wikipedia.org/wiki/Electronic_color_code)
```
Comment (rpottsoh) [inline exercises/resistor-color/description.md]: I really like this exercise!  We should nip this very minor eof character issue in the bud now so no one has to feel it's pain later.

File: exercises/resistor-color/description.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
```
Comment (petertseng) [inline exercises/resistor-color/description.md]: for reasons shown in https://webaim.org/techniques/hypertext/#screen_readers , I prefer not to have link texts such as "here". Perhaps something like "Electronic color code Wikipedia article" ?

File: exercises/resistor-color/canonical-data.json [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    {
+      "description": "Color codes",
+      "cases": [
+        {
+          "description": "Black",
+          "property": "colorCode",
+          "input": {
+            "color": "black"
+          },
+          "expected": "0"
```
Comment (rpottsoh) [inline exercises/resistor-color/canonical-data.json]: Are strings the intended output or should it be integer type?

Comment (F3PiX) [reply exercises/resistor-color/canonical-data.json]: That's a great point! It should be integers. Agree, @ErikSchierboom ?

**Comment by F3PiX** [comment]
> I am curious if you have a road map for this "series" of exercises; pun intended.

:-) 
Our focus is on the easy exercises first. So, #2 should be next (because #1 is too easy in Ruby to be a core.) 
After that, we're probably doing some other easy exercises first, with a datetime and a hash/dictionary. 
I have LOTS of ideas for exercises, and not enough time/people to implement them.

File: exercises/resistor-color/canonical-data.json (hunk 3 above)
Comment (ErikSchierboom) [reply exercises/resistor-color/canonical-data.json]: It should definitely have been integers. Fixed!

File: exercises/resistor-color/description.md (hunk 2 above)
Comment (ErikSchierboom) [reply exercises/resistor-color/description.md]: Agreed! Text has been updated.

**Pull request merged**